### ISSUE TITLE
Use libvterm's `vterm_output_set_callback`

### ISSUE THREAD

**Issue by leonerd** [neovim/neovim#9555]
Rather than using the current buffer-based logic of querying the size of buffer and then making a `read()`-like call on it, libvterm version +bzr732 adds a new function

```
typedef void VTermOutputCallback(const char *s, size_t len, void *user);
void vterm_output_set_callback(VTerm *vt, VTermOutputCallback *func, void *user);
```

which you should use in preference. Give that a pointer to some (`static`) function libvterm will invoke it when there's more output to create. You can then store that in your own buffer if you wish, or write it directly down filehandles, whatever you feel best.

This new callback API has advantage that now there's no buffer within vterm instance to get full, so fewer truncation issues.

For now I'm keeping the older buffer logic present in libvterm but eventually I'll remove it, so you'll want to have migrated this change sometime before that.

**Comment by justinmk** [comment]
Related https://github.com/neovim/neovim/issues/2514

**Comment by cryptomilk** [comment]
It would be great to get libvterm 0.2 support implemented. All the PRs which started to do that have been closed :disappointed:

**Comment by clason** [comment]
Yes. That should tell you something. Maybe you can finish the job?

**Comment by cryptomilk** [comment]
I have a minimal change for libvterm 0.2 which seems to work: https://github.com/neovim/neovim/pull/16861

**Comment by clason** [comment]
You're welcome to reopen it (rebase on master, but be aware of Windows CI failures), I don't think the other PR will get finished anytime soon.

**Comment by cryptomilk** [comment]
Looks like I can't reopen it as it has been marked as a duplicate?

**Comment by zeertzjq** [comment]
It seems GitHub won't allow you to reopen a closed PR if you force-push to it after closing it.

**Comment by cryptomilk** [comment]
Just using `vterm_output_set_callback` seems to work: https://github.com/neovim/neovim/pull/17332

**Comment by zeertzjq** [comment]
Closed by #17332
